[DocTest] Doctest for Electra Pytorch

bhadreshpsavani opened the pull request:
# What does this PR do?
Adds Doctest fo Electra Pytorch

Issue: #16292

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?


## Who can review?
@ydshieh

bhadreshpsavani committed:
Doctest for Pytorch

bhadreshpsavani committed:
Doctest for Pytorch

bhadreshpsavani committed:
Merge branch 'main' into add-doctests-electra

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_16628). All of your documentation changes will be reflected on that endpoint.

ydshieh commented on src/transformers/models/electra/modeling_electra.py:
```diff
@@ -1256,9 +1261,11 @@ def __init__(self, config):
     @add_code_sample_docstrings(
         processor_class=_TOKENIZER_FOR_DOC,
-        checkpoint=_CHECKPOINT_FOR_DOC,
+        checkpoint="bhadresh-savani/electra-base-discriminator-finetuned-conll03-english ",
```
nice!

ydshieh reviewed (commented):
Hi, @bhadreshpsavani 

It looks great, thank you for your work! I leave a few comments.
(I also saw you uploaded a checkpoint, very nice!)

bhadreshpsavani committed:
upaded model file

bhadreshpsavani committed:
upaded model file

bhadreshpsavani committed:
fixes additional space in model name

bhadreshpsavani committed:
fixes additional space in model name

bhadreshpsavani committed:
Empty commit

bhadreshpsavani committed:
Empty commit

bhadreshpsavani committed:
Merge branch 'huggingface:main' into add-doctests-electra

bhadreshpsavani committed:
Empty commit

bhadreshpsavani committed:
Empty commit

NielsRogge committed:
Fix doc example (#16448)

* Fix doc

* Make fixup

Co-authored-by: Niels Rogge <nielsrogge@nielss-mbp.home>

bhadreshpsavani commented:
Hi @ydshieh,
For pytorch its running fine,
only In case of tensorflow its giving issue,
I am sharing [notebook](https://colab.research.google.com/drive/1buzYMKgm3b8kfTcyVvnToJCFLqdnZDxM?usp=sharing) for reproducing the issue.

ydshieh commented:
> Hi @ydshieh, The issue we discussed in Issue:16292 For PyTorch its running fine, only In case of TensorFlow its giving issue, I am sharing [notebook](https://colab.research.google.com/drive/1buzYMKgm3b8kfTcyVvnToJCFLqdnZDxM?usp=sharing) for reproducing the issue.

Hi, I am not able to open it. Could you make it public?

bhadreshpsavani commented:
Hi @ydshieh,
I have updated the link

ydshieh commented:
@bhadreshpsavani Thanks. It looks like PyTorch somehow made it to run, despite the indices are wrong and out of range.
It gave strange loss value thought.

I opened a PR to address this situation

https://github.com/huggingface/transformers/pull/16648. Let's wait it being reviewed and merged to proceed.

Thank you for your patience!

lmvasque committed:
Add inputs vector to calculate metric method (#16461)

* Add inputs vector to calculate metric method

* Include inputs for evaluation metrics with backwards compatibility

* Prevent inputs create OOM issue and documentation details

* Update style and code documentation

* Fix style formatting issues

* Update files format with make style

stas00 committed:
[megatron-bert-uncased-345m] fix conversion (#16639)

sgugger committed:
Remove parent/child tests in auto model tests (#16653)

FrancescoSaverioZuppichini committed:
Updated _load_pretrained_model_low_mem to check if keys are in the state_dict (#16643)

* Updated _load_pretrained_model_low_mem to check if keys are in the stored state_dict

* update after conversions

BritneyMuller committed:
Update Support image on README.md (#16615)

* Update README.md Support Image

Updates the Support image linking to our EAP page (to give it a refresh + help avoid image fatigue).

Slack thread checking in with #open-source-internal on this update (https://huggingface.slack.com/archives/C021H1P1HKR/p1648838903316709)

* Compressed Updated Support image

* Improves Support Image Logo + Height

Updated the image based on logo + size feedback. Big thanks to Bibi for making quick edits to this image.

FrancescoSaverioZuppichini committed:
RegNet (#16188)

* base model done

* make style

* done

* added files

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Trigger doc build

* resolved conversations

* resolved conversations

* seer models

* minor changes

* minor changes

* make fixup

* glob variables

* minor changes

* fix copies

* config when possibile

* resolved conflicts

* resolved conflicts

* resolved conflicts

* CI

* conversion script for 10b param

* fixed for 10b model

* minor updates in the doc + make style

* removed unused code

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* removed unused code

* removed unused code

* updated modeling_utils from main

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <Sylvain.gugger@gmail.com>

stefan-it committed:
bert: properly mention deprecation of TF2 conversion script (#16171)

NielsRogge committed:
Add TAPEX (#16473)

* Add TapexTokenizer

* Improve docstrings and provide option to provide answer

* Remove option for pretokenized inputs

* Add TAPEX to README

* Fix copies

* Remove option for pretokenized inputs

* Initial commit: add tapex fine-tuning examples on both table-based question answering and table-based fact verification.

* - Draft a README file for running the script and introducing some background.
- Remove unused code lines in tabfact script.
- Disable the deafult `pad_to_max_length` option which is memory-consuming.

* * Support `as_target_tokenizer` function for TapexTokenizer.
* Fix the do_lower_case behaviour of TapexTokenizer.
* Add unit tests for target scenarios and cased/uncased scenarios for both source and target.

* * Replace the label BartTokenizer with TapexTokenizer's as_target_tokenizer function.
* Fix typos in tapex example README.

* * fix the evaluation script - remove the property `task_name`

* * Make the label space more clear for tabfact tasks

* * Using a new fine-tuning script for tapex-base on tabfact.

* * Remove the lowercase code outside the tokenizer - we use the tokenizer to control whether do_lower_case
* Guarantee the hyper-parameter can be run without out-of-memory on 16GB card and report the new reproduced number on wikisql

* * Remove the default tokenizer_name option.
* Provide evaluation command.

* * Support for WikiTableQuestion dataset.

* Fix a typo in README.

* * Fix the datasets's key name in WikiTableQuestions

* Run make fixup and move test to folder

* Fix quality

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Apply some more suggestions from code review

* Improve docstrings

* Overwrite failing test

* Improve comment in example scripts

* Fix rebase

* Add TAPEX to Auto mapping

* Add TAPEX to auto config mappings

* Put TAPEX higher than BART in auto mapping

* Add TAPEX to doc tests

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MBP.localdomain>
Co-authored-by: SivilTaram <qianlxc@outlook.com>
Co-authored-by: Niels Rogge <nielsrogge@nielss-mbp.home>
Co-authored-by: Suraj Patil <surajp815@gmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

johko committed:
add vit tf doctest with @add_code_sample_docstrings (#16636)

* add vit tf doctest with @add_code_sample_docstrings

* add labels string back in

Co-authored-by: Johannes Kolbe <johannes.kolbe@tech.better.team>

secsilm committed:
Fix error in doc of `DataCollatorWithPadding` (#16662)

The defalut value of `padding` in `DataCollatorWithPadding` is `True`, not `False`.

sgugger committed:
Fix style

ydshieh committed:
Fix QA sample (#16648)

* fix QA sample

* For TF_QUESTION_ANSWERING_SAMPLE

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

muellerzr committed:
Add tests for no_trainer and fix existing examples (#16656)

* Fixed some bugs involving saving during epochs
* Added tests mimicking the existing examples tests
* Added in json exporting to all `no_trainer` examples for consistency

bhadreshpsavani committed:
Merge branch 'add-doctests-electra' of https://github.com/bhadreshpsavani/transformers into add-doctests-electra

bhadreshpsavani commented:
Hi @ydshieh,

I think when I used this code and rebase at the end I messed up the commits, is there any way to remove earlier commits from this PR?

ydshieh commented:
Hi, @bhadreshpsavani 

I think close this PR and create a new one is the easiest way (considering the change in your PR is not so large).
Thank you for the effort.

BTW, I think you used `git merge` rather than `git rebase`. In general, `git rebase` could avoid this situation :-).

bhadreshpsavani commented:
I followed this,
```
git checkout main  # or `master`, depends on your local clone
git fetch upstream
git pull upstream main  # Hugging Face `transformers` renamed the default branch to `main` recently
git checkout your_working_branch_for_this_sprint
git rebase main  # or `master`
```
In the end, there was a merge conflict so i use something like `git rebase --continue` after solving the conflict and sync the changes. It got messed up!

I need to be more careful!

ydshieh commented:
> I followed this,
> 
> ```
> git checkout main  # or `master`, depends on your local clone
> git fetch upstream
> git pull upstream main  # Hugging Face `transformers` renamed the default branch to `main` recently
> git checkout your_working_branch_for_this_sprint
> git rebase main  # or `master`
> ```
> 
> In the end, there was a merge conflict so i use something like `git rebase --continue` after solving the conflict and sync the changes. It got messed up!
> 
> I need to be more careful!

OK, conflict is not easy thing.
